lib/Runtime/Debug/DiagObjectModel.cpp

```diff
@@ -3657,6 +3667,101 @@ namespace Js
+    {
+    }
+
+    BOOL RecyclablePromiseObjectWalker::GetGroupObject(ResolvedObject* pResolvedObject)
+    {
+        pResolvedObject->name = _u("[Promise]");
+        pResolvedObject->propId = Constants::NoProperty;
+        pResolvedObject->obj = instance;
+        pResolvedObject->scriptContext = scriptContext;
+        pResolvedObject->typeId = JavascriptOperators::GetTypeId(pResolvedObject->obj);
+        pResolvedObject->address = nullptr;
+
+        pResolvedObject->objectDisplay = Anew(GetArenaFromContext(scriptContext), RecyclablePromiseObjectDisplay, pResolvedObject);
+        pResolvedObject->objectDisplay->SetDefaultTypeAttribute(DBGPROP_ATTRIB_VALUE_READONLY | DBGPROP_ATTRIB_VALUE_IS_FAKE);
+        return TRUE;
+    }
+
+    BOOL RecyclablePromiseObjectWalker::Get(int i, ResolvedObject* pResolvedObject)
+    {
+        JavascriptPromise* promise = JavascriptPromise::FromVar(instance);
+
+        if (i == 0)
+        {
+            pResolvedObject->name = _u("[status]");
+
+            switch (promise->GetStatus())
+            {
+            case JavascriptPromise::PromiseStatusCode_Undefined:
+                pResolvedObject->obj = scriptContext->GetLibrary()->GetUndefinedDisplayString();
+                break;
+            case JavascriptPromise::PromiseStatusCode_Unresolved:
+                pResolvedObject->obj = scriptContext->GetLibrary()->CreateStringFromCppLiteral(_u("pending"));
+            case JavascriptPromise::PromiseStatusCode_HasResolution:
+                pResolvedObject->obj = scriptContext->GetLibrary()->CreateStringFromCppLiteral(_u("resolved"));
+                break;
+            case JavascriptPromise::PromiseStatusCode_HasRejection:
+                pResolvedObject->obj = scriptContext->GetLibrary()->CreateStringFromCppLiteral(_u("rejected"));
+                break;
+            default:
+                AssertMsg(false, "New PromiseStatusCode not handled in debugger");
+                pResolvedObject->obj = scriptContext->GetLibrary()->GetUndefinedDisplayString();
+                break;
+            }
+        }
+        else if (i == 1)
+        {
+            pResolvedObject->name = _u("[value]");
+            Var result = promise->GetResult();
+            pResolvedObject->obj = result != nullptr ? result : scriptContext->GetLibrary()->GetUndefinedDisplayString();
+        }
+        else
+        {
+            Assert(false);
+            return FALSE;
+        }
+
+        pResolvedObject->propId = Constants::NoProperty;
+        pResolvedObject->scriptContext = scriptContext;
+        pResolvedObject->typeId = JavascriptOperators::GetTypeId(pResolvedObject->obj);
+        pResolvedObject->objectDisplay = pResolvedObject->CreateDisplay();
+        pResolvedObject->objectDisplay->SetDefaultTypeAttribute(DBGPROP_ATTRIB_VALUE_READONLY | DBGPROP_ATTRIB_VALUE_IS_FAKE);
+        pResolvedObject->address = Anew(GetArenaFromContext(pResolvedObject->scriptContext),
```

Comment:
above fields ([status] and [value]) are not editable - why do you need to populate the address?